src/main/electron-normalize.ts

```diff
@@ -20,13 +20,17 @@ export const isPackaged = (() => {
 
 /** A promise that is resolved when the app is ready */
 export const whenAppReady: Promise<void> = (() => {
-  return app.isReady()
-    ? Promise.resolve()
-    : new Promise<void>((resolve) => {
-        app.once('ready', () => {
-          resolve();
+  if (app) {
+    return app.isReady()
```

Comment:
when is app not defined?

Reply:
If this code gets loaded into the Electron renderer rather than the main process.

We have code that displays a warning if you load the wrong code into the wrong process. Rollup requires tha we target `module: esnext` and it then transpiles to CJS/ESM. The issue with this is that imports/side-effects are changed/moved and these bits of code get hit before our warning code!